components/ContractViewer/index.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useState, useEffect, useCallback } from 'react'
+
+import { /*bufferToHex, Address,*/ isValidAddress } from '@ethereumjs/util'
+import { useRouter } from 'next/router'
+// import { useTheme } from 'next-themes'
+import NoSSR from 'react-no-ssr'
+
+import { solidityCompiler } from 'util/solc'
+
+import ContractCodeEditor from './ContractCodeEditor'
+import { ContractInfo, state } from './ContractState'
+import ContractTreeView from './ContractTreeView'
+
+import {
+  ResizableHandle,
+  ResizablePanel,
+  ResizablePanelGroup,
+} from '@/components/ui/resizable'
+
+function startCompilation(contract: ContractInfo) {
+  solidityCompiler.compileCode(
+    contract.code,
+    contract.etherscanInfo.CompilerVersion,
+    ['abi'], // 'ast'
+  )
+}
+
+const ContractViewer = () => {
+  const router = useRouter()
+
+  // const { theme } = useTheme()
+
+  const [status, setStatus] = useState('Loading...')
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [currentAddress, setCurrentAddress] = useState<string>('')
+  const [currentCode, setCurrentCode] = useState<string>('')
+  const [codePeekLocation, setCodePeekLocation] = useState<any>({})
+
+  // TODO: move compilation to ContractsState?
+  const onCompilationResult = (event: MessageEvent) => {
+    // console.log('onCompilationResult', event.data)
```

Comment:
Remove console.log comment. There is more commented out code too, can it go?

Reply:
why do you despise comments with such passion? comments are great!
really helps the flow of thought.
it can occasionally be trimmed, but I think obsessing about it on per-commit basis is counterproductive.